hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/FilterParams.java

```diff
@@ -64,8 +83,8 @@ public FilterParams(final Collection<TargetUpdateStatus> filterByStatus, final B
      *
      * @return {@link DistributionSet#getId()} to filter the result
      */
-    public Long getFilterByDistributionId() {
-        return filterByDistributionId;
+    public Optional<Long> getFilterByDistributionId() {
```

Comment:
I'm not 100% sure if this methods really needs to return an Optional. FilterParams is actually just a bean which exposes its fields via getters. If a field is not present I think it is fair to just return null. Just my honest opinion... :-)
Same applies to getOverdueState and getFilterBySearchText below.

Reply:
As a general rule I try to avoid returning `null` if a property of a bean is not required for constructing it - it is "optional".